ebos/eclgenericthresholdpressure.cc

```diff
@@ -248,6 +250,56 @@ getRestartVector() const
     return result;
 }
 
+template<class Grid, class GridView, class ElementMapper, class Scalar>
+void
+EclGenericThresholdPressure<Grid,GridView,ElementMapper,Scalar>::
+logPressures()
+{
+    if (!enableThresholdPressure_)
+        return;
+
+    auto lineFormat = [](unsigned i, unsigned j, double val, auto unit)
+    {
+        return fmt::format("{:4}{:>6}{:23}{:>6}{:24}{:>11.7}{:7}{}\n",
+                           " ", i, " ", j, " ", val, "", unit);
+    };
+    auto lineFormatS = [](auto s1, auto s2, auto s3)
+    {
+        return fmt::format("{:4}{:^16}{:13}{:^9}{:21}{:^18}\n",
+                           " ", s1, " ", s2, " ", s3);
+    };
+
+    std::string str = "\nLIST OF ALL NON-ZERO THRESHOLD PRESSURES\n"
+                      "----------------------------------------\n"
+                      "\n";
+    str += lineFormatS("FLOW FROM REGION", "TO REGION", "THRESHOLD PRESSURE");
+    str += lineFormatS(std::string(16, '-'), std::string(9, '-'), std::string(18, '-'));
+    const auto& simConfig = eclState_.getSimulationConfig();
+    const auto& thpres = simConfig.getThresholdPressure();
+    const auto& units = eclState_.getUnits();
+
+    for (unsigned i = 1; i <= numEquilRegions_; ++i) {
+        for (unsigned j = (thpres.irreversible() ? 1 : i); j <= numEquilRegions_; ++j) {
+            if (thpres.hasRegionBarrier(i, j)) {
+                if (thpres.hasThresholdPressure(i, j)) {
+                    str += lineFormat(i, j,
+                                      units.from_si(UnitSystem::measure::pressure,
+                                                    thpres.getThresholdPressure(j, i)),
+                                      units.name(UnitSystem::measure::pressure));
```

Comment:
I think I'd probably push the unit conversion into the `lineFormat()` helper to avoid having to type it twice.

Reply:
yeah, good point. missed to realize that opportunity during refactoring.